### ISSUE TITLE
conda dies on window resize at critical moments

### ISSUE THREAD

**Issue by bkreider** [conda/conda#7218]
**I'm submitting a...**
  - [x] bug report

## Current Behavior
Conda 3.4.34 on OSX

Conda dies horribly if the terminal is resized when connecting to repos.  I believe SIGWINCH is breaking the select call when wrapping the socket with ssl.  

### Steps to Reproduce

1. Run conda search
2. Immediately resize the terminal window multiple times. (grab a corner, resize, and let go)

```
/opt/anaconda/bin/conda search`




    Traceback (most recent call last):
      File "/opt/anaconda/lib/python2.7/site-packages/conda/exceptions.py", line 640, in conda_exception_handler
        return_value = func(*args, **kwargs)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main.py", line 140, in _main
        exit_code = args.func(args, p)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main_search.py", line 126, in execute
        execute_search(args, parser)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/cli/main_search.py", line 197, in execute_search
        unknown=args.unknown)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/common/io.py", line 174, in decorated
        return f(*args, **kwds)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/core/index.py", line 103, in get_index
        index = fetch_index(channel_priority_map, use_cache=use_cache)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/core/index.py", line 122, in fetch_index
        repodatas = collect_all_repodata(use_cache, tasks)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/core/repodata.py", line 75, in collect_all_repodata
        repodatas = _collect_repodatas_serial(use_cache, tasks)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/core/repodata.py", line 485, in _collect_repodatas_serial
        for url, schan, pri in tasks]
      File "/opt/anaconda/lib/python2.7/site-packages/conda/core/repodata.py", line 115, in func
        res = f(*args, **kwargs)
      File "/opt/anaconda/lib/python2.7/site-packages/conda/core/repodata.py", line 464, in fetch_repodata
        mod_etag_headers.get('_mod'))
      File "/opt/anaconda/lib/python2.7/site-packages/conda/core/repodata.py", line 144, in fetch_repodata_remote_request
        timeout=timeout)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/sessions.py", line 501, in get
        return self.request('GET', url, **kwargs)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/sessions.py", line 488, in request
        resp = self.send(prep, **send_kwargs)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/sessions.py", line 609, in send
        r = adapter.send(request, **kwargs)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/adapters.py", line 423, in send
        timeout=timeout
      File "/opt/anaconda/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py", line 594, in urlopen
        chunked=chunked)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py", line 350, in _make_request
        self._validate_conn(conn)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/packages/urllib3/connectionpool.py", line 835, in _validate_conn
        conn.connect()
      File "/opt/anaconda/lib/python2.7/site-packages/requests/packages/urllib3/connection.py", line 323, in connect
        ssl_context=context)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/packages/urllib3/util/ssl_.py", line 324, in ssl_wrap_socket
        return context.wrap_socket(sock, server_hostname=server_hostname)
      File "/opt/anaconda/lib/python2.7/site-packages/requests/packages/urllib3/contrib/pyopenssl.py", line 419, in wrap_socket
        rd, _, _ = select.select([sock], [], [], sock.gettimeout())
    error: (4, 'Interrupted system call')
```

## Expected Behavior

Conda should not break when the terminal window is resized.

## Environment Information
<details open><summary><code>`conda info`</code></summary><p>

```
Current conda install:

               platform : osx-64
          conda version : 4.3.34
       conda is private : False
      conda-env version : 4.3.34
    conda-build version : 2.1.17
         python version : 2.7.13.final.0
       requests version : 2.12.4
       root environment : /opt/anaconda  (writable)
    default environment : /opt/anaconda
       envs directories : /opt/anaconda/envs
                          /Users/bkreider/.conda/envs
          package cache : /opt/anaconda/pkgs
                          /Users/bkreider/.conda/pkgs
           channel URLs : https://repo.continuum.io/pkgs/pro/osx-64
                          https://repo.continuum.io/pkgs/pro/noarch
                          https://repo.continuum.io/pkgs/free/osx-64
                          https://repo.continuum.io/pkgs/free/noarch
                          https://repo.continuum.io/pkgs/main/osx-64
                          https://repo.continuum.io/pkgs/main/noarch
                          https://conda.anaconda.org/wakari/osx-64
                          https://conda.anaconda.org/wakari/noarch
            config file : /opt/anaconda/.condarc
             netrc file : None
           offline mode : False
             user-agent : conda/4.3.34 requests/2.12.4 CPython/2.7.13 Darwin/16.7.0 OSX/10.12.6    
                UID:GID : 502:20

```
</p></details>

<details open><summary><code>`conda config --show-sources`</code></summary><p>

```
~$ conda config --show-sources
==> /opt/anaconda/.condarc <==
channels:
  - https://repo.continuum.io/pkgs/pro
  - https://repo.continuum.io/pkgs/free
  - https://repo.continuum.io/pkgs/main
  - https://conda.anaconda.org/wakari
default_channels:
  - https://repo.continuum.io/pkgs/free
channel_alias: https://conda.anaconda.org
show_channel_urls: True
binstar_upload: True
```
</p></details>

**Comment by pzwang** [comment]
Seems like the same problem as this:  https://github.com/certbot/certbot/issues/4871
There, they claim the upstream bug in urllib3 is https://github.com/urllib3/urllib3/issues/700,
which was merged in November 2016: https://github.com/urllib3/urllib3/pull/1001

What version of urllib3 did you see this problem with?

**Comment by kalefranz** [comment]
requests vendors a very old version of urllib3 :(

**Comment by kalefranz** [comment]
And looks like we have here:
requests version : 2.12.4

**Comment by kalefranz** [comment]
requests 2.12.4 vendors urllib3 1.19.1
https://github.com/requests/requests/blob/v2.12.4/requests/packages/urllib3/__init__.py#L35

**Comment by kalefranz** [comment]
urllib3 1.19.1 was published on Nov 16, 2016
https://github.com/urllib3/urllib3/commits/1.19.1

**Comment by kalefranz** [comment]
urllib3/urllib3#1001 wasn't included until 1.20.0.
https://github.com/urllib3/urllib3/blob/master/CHANGES.rst#120-2017-01-19

**Comment by kalefranz** [comment]
Just need to update requests.
